src/Components/NotFoundErrorPage.js

```diff
@@ -4,10 +4,13 @@ import { Helmet } from "react-helmet-async";
 
 class NotFoundErrorPage extends React.Component {
   componentDidMount() {
-    const path = window.location.pathname;
+    const statusCode = "status-code=404-not-found";
+    const { search } = window.location;
 
-    if (path !== "/404") {
-      window.location.replace(`/404?path=${encodeURIComponent(path)}`);
+    if (!Scrivito.isEditorLoggedIn() && !search.includes(statusCode)) {
```

Comment:
* [ ] FYI if we make the `isEditorLoggedIn` a more early guard, and/or the other check a guard, the readability may improve - but I haven't tried, so I may be wrong.

Reply:
Is that really better?

```js
    if (Scrivito.isEditorLoggedIn()) {
      return;
    }
    const statusCode = "status-code=404-not-found";
    const { search } = window.location;
    if (search.includes(statusCode)) {
      return;
    }

    window.location.search = search
      ? `${search}&${statusCode}`
      : `?${statusCode}`;
```

One line if/return statements are currently not allowed by the eslint rules.

Reply:
No, it isn't. The latest version of `componentDidMount` became more readable already.

* [ ] FYI One more idea is making the `const statusCode = "status-code=404-not-found";`  an actual toplevel const. No need to put it in the function body.